File: backend/CinebhApp/src/main/java/com/atlantbh/internship/CinebhApp/domain/Movie.java
package com.atlantbh.internship.CinebhApp.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Movie {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String pgRating;
    private String language;
    private String duration;
    private Date startDate;
    private Date endDate;
    private String trailer;
    private String director;
    private String synopsis;
    private MovieStatus status;
}

File: frontend/src/models/Movie.ts
import { MovieGenre } from "models/MovieGenre";
import { MovieImage } from "models/MovieImage";
import { MovieStatus } from "models/MovieStatus";
import { Projection } from "models/Projection";

export type Movie = {
  id: number;
  name: string;
  pgRating: string;
  language: string;
  duration: string;
  startDate: string;
  endDate: string;
  trailer: string;
  tmdbId: string;
  synopsis: string;
  status: MovieStatus;
  genres: MovieGenre[];
  images: MovieImage[];
  projections: Projection[];
};
